jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/LambdaConfigPanel.kt

```diff
@@ -80,20 +102,22 @@ class LambdaConfigPanel(private val project: Project) : JPanel(BorderLayout()) {
         iamRole = builder().resource(IamResources.LIST_LAMBDA_ROLES).awsConnection(project).build()
     }
 
-    fun setXrayControlVisibility(visible: Boolean) {
+    private fun setXrayControlVisibility(visible: Boolean) {
         xrayEnabled.isVisible = visible
         if (!visible) {
             xrayEnabled.isSelected = false
         }
     }
 
-    fun validatePanel(): ValidationInfo? {
-        handlerPanel.handler.text.nullize(true)
-            ?: return handlerPanel.handler.validationInfo(message("lambda.upload_validation.handler"))
-
+    fun validatePanel(handlerMustExist: Boolean): ValidationInfo? {
+        // Check runtime first, since handler panel needs it to be there
         runtimeModel.selectedItem
             ?: return runtime.validationInfo(message("lambda.upload_validation.runtime"))
```

Comment:
extreme nit: this looks like it was formatted before we made our line size longer so some of the spacing in this file is weird

Reply:
Updated